sphinx/themes/basic/static/searchtools.js

```diff
@@ -328,10 +328,14 @@ const Search = {
     for (const [title, foundTitles] of Object.entries(allTitles)) {
       if (title.toLowerCase().trim().includes(queryLower) && (queryLower.length >= title.length/2)) {
         for (const [file, id] of foundTitles) {
-          let score = Math.round(100 * queryLower.length / title.length)
+          let isMainTitle = titles[file] === title
+          // score these a little bit above document matches, with more of a boost
+          // for main document titles
+          let baseScore = Scorer.title + (isMainTitle ? 2 : 1)
+          let score = Math.round(baseScore * queryLower.length / title.length)
```

Comment:
```suggestion
          let baseScore = (isMainTitle ? Scorer.title : Scorer.partialTitle) + 1
```

Got more relevant results by doing the above, but it feels even more hacky. Also comes at the cost of pushing subtitle matches even further down.

<img width="1412" alt="image" src="https://github.com/sphinx-doc/sphinx/assets/20569/3fcc7a2f-330e-4f4b-9e2e-19765e89973b">